### ISSUE TITLE
Filter fields

### ISSUE THREAD

**Issue by theo2f** [wfletcher/EasyGIS.NET#73]
The DbfReader class have the method IndexOf to filter but it's marked as obsolete.
I would like to know what's the alternative to this method.

Other thing is that the DbfReader class is sealed. I can't create a new class and inhiret the DbfReader class to create my own filter method.
I want to filter with many conditions options like partial text or value >, <, StartsWith or EndsWith...
Actually I have a filter with all of this using Sharpmap library, but it's not working with very large files. Since DbfReader class is working fine with these files, I'm working on it to solve this issue.

![image](https://user-images.githubusercontent.com/5304192/180003136-6a29baad-0dab-4341-94bb-4522e90c9739.png)

**Comment by wfletcher** [comment]
The method has been marked as obsolete for years. To be honest I marked it with this attribute as I wasn’t happy with the implementation and wasn’t sure whether I was going to keep it. The main issue I have is that the method is indexing a single field in memory using a sorted List<string> which can be slow to create  when using  large dbf files.

If you think it is useful I’m happy to “unseal” the class so you can create a derived class.

**Comment by theo2f** [comment]
I installed and tested some libraries that work with DBF.
The method I used to measure the performance was to list all the fields in the table, because only then would I be able to measure it in an equivalent way, since in the other components a DataTable is always delivered with all the values.
Incredibly, listing all records for all fields through the DbfReader class did not take significantly longer than the other libraries. Honestly, I don't understand how this was possible.
I changed my software code to fetch data via DbfReader class and the time to fetch 3 fields and manipulate them changed from 0.380ms to 0.435ms. I'll keep using DbfReader class.
